react/AddProductBtn.tsx

```diff
@@ -351,6 +351,23 @@ const AddBtn: FC<AddBtnProps> = ({ toastURL = '/account/#wishlist' }) => {
     }
   }
 
+  if (
+    data?.checkList?.inList !== true &&
+    productCheck[productId] === undefined &&
+    wishListed.find(
+      (item: any) => item.productId === productId && item.sku === sku
+    ) !== undefined
+  ) {
+    wishListed = wishListed.filter(
+      (item: any) => item.productId !== productId && item.sku !== sku
+    )
+    saveToLocalStorageItem(wishListed)
+    setState({
+      ...state,
+      isWishlistPage: false,
+    })
+  }
+
```

Comment:
```suggestion
  
  wishListed = wishListed.filter(
      (item: any) => item.productId !== productId && item.sku !== sku
  )
  
  if (
    data?.checkList?.inList !== true &&
    productCheck[productId] === undefined &&
    wishListed.length > 0
  ) {    
    saveToLocalStorageItem(wishListed)
    setState({
      ...state,
      isWishlistPage: false,
    })
  }

```

Take a look at this suggestion, first, you filter the array, and then you check if the length is greater than 0. 

**Probably my suggestion doesn't have a valid eslint check.**

Reply:
Thank you for the suggestion. But the `wishListed` could originally have more than one item in the list. 
And we want to use the filter to remove the item only under the condition that we find the item in the `wishListed` and the inList in false(whish from the network).

Reply:
Thank you for the suggestion. But the `wishListed` could originally have more than one item in the list.
And we want to use the filter to remove the item only under the condition that we find the item in the `wishListed` and the `inList` in false(whish from the network).